### PR TITLE
fix (paginator): fix panic when isNil(...) is called with an arrary

### PR DESCRIPTION
**Pull request by jimlambrt**

### CODE REVIEW

File: paginator/is_nil_test.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package paginator
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_IsNil(t *testing.T) {
```
Comment (pilagod) [inline paginator/is_nil_test.go:10]: [minor]

I think we could rename this file name to `paginator_utils_test.go` to keep the consistency between other test files.

And this test case maybe could be called `TestIsNil` to conform to golang naming convention.

File: paginator/is_nil_test.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	tc := []struct {
+		i    interface{}
+		want bool
+	}{
+		{i: &testError{}, want: false},
+		{i: testError{}, want: false},
+		{i: &map[string]struct{}{}, want: false},
+		{i: map[string]struct{}{}, want: false},
+		{i: [1]string{}, want: false},
+		{i: &[1]string{}, want: false},
```
Comment (pilagod) [inline paginator/is_nil_test.go:30]: Just wondering that are these two cases causing the old version to break?

**Review by pilagod — COMMENTED** [review]
Thanks @jimlambrt for the finding 😃 

I didn't aware there is problem for array value since most of my usage is using slice 😂

Most of the changes look good to me, just leaving some minor points 👍

File: paginator/is_nil_test.go (hunk 2 above)
Comment (jimlambrt) [reply paginator/is_nil_test.go:30]: Just the first test: `{i: [1]string{}, want: false}` causes the panic.  The next test is a ptr to array and that's okay with the original version.

**Comment by jimlambrt** [comment]
@pilagod - we bumped into this issue because we're using [google's uuid in our models](https://github.com/google/uuid/blob/44b5fee7c49cf3bcdf723f106b36d56ef13ccc88/uuid.go#L20) and it's defined as a `type UUID [16]byte`

**Comment by pilagod** [comment]
Thanks for the finding. I think this is a quite common case this library should take care of. 

LGTM. I am going to merge it 💪

**Comment by pilagod** [comment]
I have published [v2.3.1](https://github.com/pilagod/gorm-cursor-paginator/releases/tag/v2.3.1) and can be upgrade to it for this fix 💪

**Pull request merged**